### PR TITLE
Added handle computer turn functionality and System Integration

### LINKED ISSUE #15: Player Module - Computer Player - Request Card
*request_card_computer()
*algorithm_for_computer()

### LINKED ISSUE #19: Player Module - Computer Player - Algorithm
* algorithm_for_computer

### PR DESCRIPTION
**Pull request by MiaYWei**
* Added handle computer turn functionality 
* Code cleanup by removing unused functions
* Added game winner determination
* System integration for release 1

### CODE REVIEW

**Review by aditibhardwaj98 — APPROVED** [review]

**Review by HongBo713 — APPROVED** [review]

File: src/cards_management.c [hunk 1]
```diff
@@ -73,15 +67,21 @@ int initialize_cards(void)
...
+            result += add_card_at_end(g_draw_pile, card);
         }
     }
 
+    //shuffle_cards();
+
     g_discard_pile = (Deck_t *)malloc(sizeof(Deck_t));
     if (g_discard_pile == NULL)
     {
         printf("Unable to allocate memory to initizlize discard pile.");
```
Comment (krithigasr) [inline src/cards_management.c]: initialise* Typo

File: src/cards_management.c [hunk 2]
```diff
@@ -505,7 +347,7 @@ int draw_cards(int num_draw_cards, PlayerType_e player)
         }
 
         draw_card = remove_first_card_at_beginning(&g_draw_pile);
-        result = add_card_at_beginning(&g_players[player].cards_on_hand, draw_card->card);
+        result = add_card_at_end(g_players[player].cards_on_hand, draw_card->card);
```
Comment (krithigasr) [inline src/cards_management.c:374]: What's the difference between adding card at the and and at the beginning here?

File: src/cards_management.c [hunk 3]
```diff
@@ -558,3 +390,94 @@ void initialize_card_on_table(void)
...
+ * @brief Remove a particular card from the given deck
+ * Can be used to discard a particular card - during each turn
+ * Works for both the computer as well as human deck of cards 
+ * 
+ * @param pp_head Pointer to the head of the deck from which card has to be removed
+ * @param card Card to be removed
+ * @return true If the card is been removed successfully
+ * @return false If the removing of card fails
+ */
+bool remove_card_from_deck(Deck_t** pp_head, const Card_t card)
```
Comment (krithigasr) [inline src/cards_management.c:428]: Will it not be easy to track errors if the functions return values from ret_types.h ? Like the function was earlier defined.

File: src/game.c [hunk 4]
```diff
@@ -13,28 +21,42 @@ int initialize_game(void);
...
-        int key_q = 113;  //ASCII code for "q"    
-        while (1) {
-            if ((key_Q == _getch()) || (key_q == _getch())) {                 
-                printf("Human Player request to quit the game.\n");
-                if (!confirm_exit()) {
-                    printf("Game is continuing...\n");
-                }
-            }
+    printf("%s player starts the game.\n\n", PLAYER_TYPE_STRING[g_player_on_turn]);
+    printf("Game is running....\n");
```
Comment (krithigasr) [inline src/game.c]: Can you change this to "Game begins now" ?

File: src/game.c [hunk 5]
```diff
@@ -13,28 +21,42 @@ int initialize_game(void);
...
 {
     bool is_confirmed = false;
-    initialize_game();
-    printf("Latest Discard Card is: (%s, %s).\n", CARD_COLOR_STRING[g_card_on_table.color], CARD_NAME_STRING[g_card_on_table.name]);
+    if (0 != initialize_game())
+    {
+        printf("Initialize game failed.\n ");
+        return;
+    }
+    printf("\nLatest Discard Card is: (%s, %s).\n", CARD_COLOR_STRING[g_card_on_table.color], CARD_NAME_STRING[g_card_on_table.name]);
```
Comment (krithigasr) [inline src/game.c]: Can you please change this to "Card on the table now:"?

File: src/game.c [hunk 6]
```diff
@@ -74,4 +99,113 @@ int initialize_game(void)
...
+        ret = true;
+    }
+    return ret;
+}
+
+/**
+ * @brief his function handles the functionality to support computer player's turn 
+ * 
+ */
+void handle_computer_turn(void)
```
Comment (krithigasr) [inline src/game.c:149]: What happens when the computer doesn't have a playable card and needs card from the draw pile?

File: src/cards_management.c (hunk 1 above)
Comment (MiaYWei) [reply src/cards_management.c]: Fixed: initialize

File: src/cards_management.c (hunk 2 above)
Comment (MiaYWei) [reply src/cards_management.c:374]: Add_card_at_end doesn't work. I reverted the code here.

File: src/cards_management.c (hunk 3 above)
Comment (MiaYWei) [reply src/cards_management.c:428]: If you want, you can make the related change. I prefer my current implementation.

File: src/game.c (hunk 6 above)
Comment (MiaYWei) [reply src/game.c:149]: If computer doesn't not have a playable card on hand, then will draw a card from the draw pile, as it is implemented in the current code.

File: src/cards_management.c (hunk 3 above)
Comment (krithigasr) [reply src/cards_management.c:428]: Sure I'll look into that.

**Review by krithigasr — APPROVED** [review]

**Pull request merged**